src/Sulu/Component/Cache/MemoizeTwigExtensionTrait.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Cache;
+
+/**
+ * Provides functionality to convert twig functions.
+ */
+trait MemoizeTwigExtensionTrait
+{
+    /**
+     * Convert simple twig functions to use a new context.
+     *
+     * @param \Twig_SimpleFunction[] $functions
+     * @param mixed $context
+     *
+     * @return \Twig_SimpleFunction[]
+     */
+    protected function convertTwigFunctions(array $functions, $context)
```

Comment:
Hm... I am not sure if I wouldn't delete that method completely and move these lines to `getFunctions`... Or at least I would make that method `private`, because it is not supposed to be called directly. If it can't be called it is less likely that we'll do a BC break in the future :-)